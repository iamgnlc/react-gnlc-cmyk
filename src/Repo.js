import React from "react"

const Repo = React.memo(() => {
  const url = "https://github.com/iamgnlc/react-gnlc-cmyk"
  return (
    <a className="repo" href={url} target="_blank" rel="noopener noreferrer">
      {url}
    </a>
  )
})

export default Repo
