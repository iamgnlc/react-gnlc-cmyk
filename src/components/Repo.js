import React, { Component } from "react"

class Repo extends Component {
  state = { show: null }

  keys = [17, 91, 93, 224]

  handleKeyDown = (event) => {
    this.keys.includes(event.keyCode) && this.setState({ show: true })
  }

  handleKeyUp = (event) => {
    this.keys.includes(event.keyCode) && this.setState({ show: false })
  }

  componentDidMount() {
    document.addEventListener("keydown", this.handleKeyDown)
    document.addEventListener("keyup", this.handleKeyUp)
  }

  componentWillUnmount() {
    document.removeEventListener("keydown", this.handleKeyDown)
    document.removeEventListener("keyup", this.handleKeyUp)
  }

  render() {
    let { show } = this.state
    return (
      <div className={`repo ${!show ? "hide" : ""}`}>
        {show && (
          <a
            href={process.env.RAZZLE_REPO_URL}
            target="_blank"
            rel="noopener noreferrer"
          >
            {process.env.RAZZLE_REPO_URL}
          </a>
        )}
      </div>
    )
  }
}

export default Repo
