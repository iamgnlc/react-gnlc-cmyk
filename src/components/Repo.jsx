import React, { useState, useEffect, useCallback } from 'react';

const Repo = React.memo(() => {
  const [show, setShow] = useState(false);

  const keys = [17, 91, 93, 224];

  const handleKeyDown = useCallback(
    (event) => {
      keys.includes(event.keyCode) && setShow(true);
    },
    [keys],
  );

  const handleKeyUp = useCallback(
    (event) => {
      keys.includes(event.keyCode) && setShow(false);
    },
    [keys],
  );

  const addEvent = useCallback(() => {
    document.addEventListener('keydown', handleKeyDown);
    document.addEventListener('keyup', handleKeyUp);
  }, [handleKeyDown, handleKeyUp]);

  const removeEvent = useCallback(() => {
    document.removeEventListener('keydown', handleKeyDown);
    document.removeEventListener('keyup', handleKeyUp);
  }, [handleKeyDown, handleKeyUp]);

  useEffect(() => {
    addEvent();
    return () => {
      removeEvent();
    };
  }, [addEvent, removeEvent]);

  return (
    <div className={`repo ${!show ? 'hide' : ''}`.trim()}>
      {show && (
        <a
          href={process.env.REACT_APP_REPO_URL}
          target="_blank"
          rel="noopener noreferrer"
        >
          {process.env.REACT_APP_REPO_URL}
        </a>
      )}
    </div>
  );
});

Repo.displayName = 'Repo';

export default Repo;
