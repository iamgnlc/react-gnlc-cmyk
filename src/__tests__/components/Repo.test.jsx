import React from 'react';
import { render } from '@testing-library/react';
import { act } from 'react-dom/test-utils';

import Repo from '../../components/Repo';

describe('Repo', () => {
  beforeEach(() => {
    process.env.REACT_APP_REPO_URL = 'github url';
  });

  it('should render correctly without showing', () => {
    const view = render(<Repo />);

    act(() => {
      const event = new KeyboardEvent('keyup', { keyCode: 93 });
      document.dispatchEvent(event);
    });

    expect(view.container).toMatchSnapshot();
  });

  it('should render correctly showing', () => {
    const view = render(<Repo />);

    act(() => {
      const event = new KeyboardEvent('keydown', { keyCode: 93 });
      document.dispatchEvent(event);
    });

    expect(view.container).toMatchSnapshot();
  });
});
