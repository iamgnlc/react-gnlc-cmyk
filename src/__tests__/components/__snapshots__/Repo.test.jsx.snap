// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repo should render correctly showing 1`] = `
<div>
  <div
    class="repo"
  >
    <a
      href="github url"
      rel="noopener noreferrer"
      target="_blank"
    >
      github url
    </a>
  </div>
</div>
`;

exports[`Repo should render correctly without showing 1`] = `
<div>
  <div
    class="repo hide"
  />
</div>
`;
